contracts/eosio.system/include/eosio.system/eosio.system.hpp

```diff
@@ -389,6 +389,19 @@ namespace eosiosystem {
 
    typedef eosio::multi_index< "acctdenylist"_n, account_name_blacklist >  account_name_blacklist_table;
 
+   // Store hash values allowing account blacklist names to be added to `account_name_blacklist_table`
+   struct [[eosio::table("acctdenyhash"), eosio::contract("eosio.system")]] deny_hash {
+      uint64_t     id;                       // automatically generated ID for the key in the table
+      checksum256  hash;                     // sha256 hash computed over a `vector<name>` and allowing
+                                             // said vector to be added to `account_name_blacklist_table`
+      uint64_t     primary_key() const { return id; }
+      checksum256  by_hash()     const { return hash; }
```

Comment:
Would this be better as `const checksum256& by_hash() const`  ?

Reply:
Probably makes little difference I think. This is called once when adding the row.